src/abstract/BaseTrigger.sol

```
@@ -49,19 +55,21 @@ abstract contract BaseTrigger is ICState, IBaseTrigger {
     return sets.length;
   }
 
-  /// @dev Call this method to update Set addresses after deploy.
-  function addSet(ISet _set) external {
+  /// @dev Call this method to update Set addresses after deploy. Returns false if the trigger has not been acknowledged.
+  function addSet(ISet _set) external returns (bool) {
     if (msg.sender != address(manager)) revert Unauthorized();
+    if (!acknowledged) return false;
```

I think we should revert here. Doing so gives two possible results for a well-behaved trigger implementation:
1. `acknowledged = true`, we return `true`
2. `acknowledged = false` / other failure modes, we revert

This is analogous to ERC-20 methods: they return a bool, but in reality its for a similar purpose as this, because most tokens will revert on failure instead of returning true.

(Of course in the manager we'd still need to validate that `true` is returned, because someone can still return false instead of reverting, but typically best-practice is to revert if there's a problem)

---

like this right? [125bfce](https://github.com/Cozy-Finance/cozy-v2-triggers/pull/18/commits/125bfcee0917885c2c0729761aa1593053f61bfc)

---

lgtm, thanks!